### DISCUSSION TITLE
It's possiable to use this project to sync blocks from Substrate-Based chain ?

### DISCUSSION THREAD

**Discussion by g302ge** [paritytech/smoldot#2963] [General]
Recently, I am trying to implementation a polkadot chain data analysis system. In this system I need sync blocks from chain, and analysis its transaction Also insert into some OLAP database. So if I can extract some transaction details using this crate ?

**Comment by tomaka** [comment]
It depends on exactly what you need.
Beware that only the light client is officially supported. Anything else might change and break at any time.

**Reply by g302ge** [reply]
Currently, the full node in smoldot is imcomplete not. If there some future plan to complete this crate let the light client can work with the pallets ?
